### DISCUSSION TITLE
How to create the connection between 2 agents and exchange the credentials

### DISCUSSION THREAD

**Discussion by thomas-tran** [decentralized-identity/veramo#326] [Q&A]
How to create the connection between 2 agents and exchange the credentials? For example, Alice's agent on the mobile app and Bob's agent on the cloud.

**Comment by jasheal** [comment]
@thomas-tran we will have a full tutorial on this soon but the basic version is:

- Create a VC, Presentation, or SDR, with `Agent 1`
- POST the message to `Agent 2` `/handleMessage` endpoint

Hope that helps!

**Reply by thomas-tran** [reply]
Can you explain more detail? For example, 
Agent 1 is a mobile agent built in iOS device app. 
Agent 2 is a mobile agent build in Android device app. 

How to establish connection between those 2 agents? 
If you are using the cloud agent then I will know the url for /handleMessage endpoint. However, for mobile app how do I know the endpoint /handleMessage for the agent?

**Reply by jasheal** [reply]
For the most part Veramo doesn't impose anything on your transport layer although we may provide some examples. The credential is just a piece of data so it can be sent like any other data depending on your use case. You could represent it as a QR code and scan it with the other app or save it as a file extension and email or send it over bluetooth. There is also WalletConnect that will allow connecting a mobile wallet to desktop application.

Also, note for cloud agent interactions using `/handleMessage` is more of an internal API. You would use the sendDIDComm endpoint which will figure out the endpoint based on the DID doc ~ (https://veramo-agent.herokuapp.com/api-docs/#/default/sendMessageDIDCommAlpha1)

/handleMessage can however be used internally on your node server app or React Native App.
